numpy/ma/extras.py

```diff
@@ -638,6 +638,15 @@ def median(a, axis=None, out=None, overwrite_input=False, keepdims=False):
     else:
         return r
 
+def _median1D(data):
+    counts = filled(count(data), 0)
+    (idx, rmd) = divmod(counts, 2)
+    if rmd:
+        choice = slice(idx, idx + 1)
+    else:
+        choice = slice(idx - 1, idx + 1)
+    return data[choice].mean(0)
```

Comment:
Note that this code is just pasted back from the old (pre #4760) version.

Actually, I have no idea why they used the "filled(..., 0)" at all:
If input data is 1d, wouldn't count() always return a scalar int?
(possibly some edge case related to the usage of _median1D under apply_along_axis, but I can't think of any).

I'd think that the whole thing could be written as:
```python                                                             
idx, is_odd = divmod(count(data), 2)
return data[idx - is_odd:idx + 1].mean()
```
and with 2 lines of code, we get even more incentive to inline this as @ahaldane suggested bellow...

I will commit this as a separate changeset, so you can cherrypick if you find the original version more robust (after all - it had already been active in many older versions of numpy).